### ISSUE TITLE
read replica / master-slave support in mysq2 npm package

### ISSUE THREAD

**Issue by rewa15** [sidorares/node-mysql2#1461]
Is there any option to add read replicas / master and slave connections using mysql2 npm package while creating a connection pool?

As we have something like this in sequelize -

`const sequelize = new Sequelize(mysql.database, mysql.user, mysql.password, {
  host: mysql.host,
  port: mysql.port,
  timezone: mysql.timezone,
  dialect: 'mysql',
  define: {
    freezeTableName: true,
  },
  replication: {
    read: [
      { host: mysql.replica1Host },
      { host: mysql.replica2Host },
    ],
    write: { host: mysql.host },
  },
  pool: {
    max: mysql.connectionLimit,
    acquire: mysql.acquireTimeout,
    idle: 30000,
  },
});`

Cannot find similar support in mysql2 npm documentation.

**Comment by sidorares** [comment]
hi @rewa15 , see answer in https://github.com/sidorares/node-mysql2/issues/1444

TLDR: mysql2 does not try to interpret sql query semantics. Sequlize can do that due to the level where it sits allows better to understand if resulting query is read or write

**Comment by rewa15** [comment]
Thanks @sidorares for the detailed context.
